doc/guides/maintaining-the-build-files.md

```diff
@@ -0,0 +1,386 @@
+# Maintaining the Build files
+
+This document explains how to maintain the build files in the codebase.
+
+## Overview
+
+On how to build the Node.js core, see [Building Node.js](../../BUILDING.md).
+
+There are three main build files that will be directly run to build Node.js:
+
+On Windows:
+
+- vcbuild.bat: A Windows Batch Script that locates build tools, provides
+  a subset of the targets avilable in the [Makefile](#makefile), and a few
+  targets of its own. This script also calls [configure](#configure) before
+  building the Node.js binary. For a detailed guide on this script,
+  see [vcbuild.bat](#vcbuild.bat).
+
+On unix-like systems:
+
+- configure: A (mostly) Python 2 script that accepts a set of options,
+  configure and prepare the build using gyp. For a detailed guide on
+  this script, see [configure](#configure).
+- Makefile: A makefile that can be run with BSD make or GNU make. It
+  provides a set of targets that build and test the Node.js
+  binary, produce releases and documentation, and interact with the
+  CI to run benchamarks or tests. For a detailed guide on this file,
+  see [Makefile](#makefile)
+
+## vcbuild.bat
+
+To see the help text, run `.\vcbuild help`. Consider updating this file
+when you need to update the build and testing process on Windows.
+
+```
+vcbuild.bat [debug/release] [msi]
+[test/test-ci/test-all/test-addons/test-addons-napi/test-internet/test-pummel/
+test-simple/test-message/test-gc/test-tick-processor/test-known-issues/
+test-node-inspect/test-check-deopts/test-npm/test-async-hooks/test-v8/
+test-v8-intl/test-v8-benchmarks/test-v8-all]
+[ignore-flaky] [static/dll] [noprojgen] [small-icu/full-icu/without-intl]
+[nobuild] [nosnapshot] [noetw] [noperfctr] [licensetf] [sign] [ia32/x86/x64]
+[vs2015/vs2017] [download-all] [enable-vtune] [lint/lint-ci/lint-js/lint-js-ci]
+[package] [build-release] [upload] [no-NODE-OPTIONS]
+[link-module path-to-module] [debug-http2] [debug-nghttp2] [clean]
+
+Examples:
+  vcbuild.bat                          : builds release build
+  vcbuild.bat debug                    : builds debug build
+  vcbuild.bat release msi              : builds release build and MSI installer package
+  vcbuild.bat test                     : builds debug build and runs tests
+  vcbuild.bat build-release            : builds the release distribution as used by nodejs.org
+  vcbuild.bat enable-vtune             : builds nodejs with Intel VTune profiling support to profile JavaScript
+  vcbuild.bat link-module my_module.js : bundles my_module as built-in module
+  vcbuild.bat lint                     : runs the C++ and JavaScript linter
+```
+
+## configure
+
+To see the help text, run `./configure --help`(unix-like)
+or `.\configure --help` (Windows). Consider updating this file
+when you need to update the configuration process.
+
+```
```

Comment:
I am not sure if pasting the help text here is useful..although it does give you an idea about what it does if you don't know what it is.

Reply:
I think the above paragraph is enough. Including the text here will only drift from the actual help text over time.

Reply:
Agreed with @richardlau , don't think we should duplicate the `./configure --help` output.